xtable-api/src/main/java/org/apache/xtable/conversion/TableSyncConfig.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.apache.xtable.conversion;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import com.google.common.base.Preconditions;
+
+import org.apache.xtable.model.sync.SyncMode;
+
+@Value
+public class TableSyncConfig {
```

Comment:
In one of the changes, we adopted "conversion" as the standard terminology, for e.g ConversionTarget, ConversionSource. Would it be more appropriate to name this class ConversionConfig?

Reply:
Sounds good, updated to ConversionConfig